packages/http-server-javascript/src/http/server/index.ts

```diff
@@ -453,7 +453,7 @@ function* emitQueryParamBinding(
   yield `const ${nameCase.camelCase} = __query_params.get(${JSON.stringify(parameter.name)}) ?? undefined;`;
 
   if (!parameter.param.optional) {
-    yield `if (${nameCase.camelCase} === null) {`;
+    yield `if (!${nameCase.camelCase}) {`;
```

Comment:
I think this is actually going to always fail. In the generated code you have this check:
```ts
const __query_params = new URLSearchParams(
    request.url!.split("?", 1)[1] ?? "",
  );
```
`.split(string, 1)` will return the 1st result, which will be whatever is on the left-hand side of the `?`. That would be the path.

I just did a quick test with this tsp:
```tsp
import "@typespec/http";

using Http;

@service
namespace Sample;

@get op foo(@query something: int8): {
  @statusCode statusCode: 200;
};
```
When I make an http request to `http://localhost:3333/?something=42`, `something` isn't part of the splitted url so ends up being `undefined`